src/core/lib/transport/metadata.cc

```diff
@@ -482,3 +479,20 @@ void grpc_mdelem_do_unref(grpc_mdelem gmd DEBUG_ARGS) {
     }
   }
 }
+
+void grpc_mdelem_finalize_unref(grpc_mdelem_data_storage storage, void* ptr,
```

Comment:
Why is "finalize" in the name? I think I know why...if it's because this is called for the last ref, then maybe "grpc_mdelem_final_unref" would be clearer?

Reply:
I think a good name could be: grpc_mdelem_on_final_unref(). Will change it.